pulsar-broker-common/src/main/java/org/apache/pulsar/broker/authentication/AuthenticationService.java

```diff
@@ -99,18 +99,38 @@ public String authenticate(AuthenticationDataSource authData, String authMethodN
     }
 
     public String authenticateHttpRequest(HttpServletRequest request) throws AuthenticationException {
-        // Try to validate with any configured provider
         AuthenticationException authenticationException = null;
         AuthenticationDataSource authData = new AuthenticationDataHttps(request);
-        for (AuthenticationProvider provider : providers.values()) {
+        String authMethodName = request.getHeader("Auth-Method-Name");
+
+        if (authMethodName != null) {
+            AuthenticationProvider providerToUse = providers
+                    .values()
+                    .parallelStream()
+                    .filter(provider -> provider.getAuthMethodName().equals(authMethodName))
+                    .findAny()
```

Comment:
`AuthenticationService` should build a map in the constructor with all the available provider indexed by name

Reply:
`providers` is already a Map built in the constructor. Not sure to understand your statement?

Reply:
Ok, so we should be able to just do `providers.get(authMethodName)` instead of iterating over the items of the map.

Reply:
Oh ok, so try catch.

Reply:
Thanks for the suggestion I thought get from Map could throws Exception is not find :D.